webpack/assets/javascripts/foreman_advanced_fields.js

```diff
@@ -0,0 +1,20 @@
+/* eslint-disable func-names */
+import $ from 'jquery';
+
+export function initAdvancedFields() {
+  $('a.advanced_fields_switch').each(function () {
+    const field = $(this);
+    field.on('click', updateAdvancedFields);
+  });
+}
+
+function updateAdvancedFields() {
+  const switcher = $('a.advanced_fields_switch');
+  const original = switcher.html();
+  switcher.html(switcher.data('alternativeLabel'));
+  switcher.data('alternativeLabel', original);
+
+  switcher.siblings('i').toggleClass('fa-angle-right').toggleClass('fa-angle-down');
```

Comment:
Nitpick: The risk is minimal but I'd be more specific with the selector here: `i.fa`.
Just to prevent selecting more elements if we add them in future.

Reply:
I think you can write
```
.toggleClass('fa-angle-right fa-angle-down')
```
but I'm not sure it's more readable. Mentioning just for consideration.

Reply:
if you don't mind, I'll keep that separate as I find it easier to read